### DISCUSSION TITLE
What is the exactly relations between versions of setuptools and python?

### DISCUSSION THREAD

**Discussion by LuffysMan** [pypa/setuptools#3858] [Q&A]
For example, can setuptools67.4.0 works correctly on python 3.7.9?

**Comment by abravalheri** [comment]
Hi @LuffysMan, for each version of setuptools, the minimum Python version via the `Requires-Python` core metadata. This can be read in the [`setuptools-*.dist-info/METADATA` file contained by the setuptools wheel](https://inspector.pypi.io/project/setuptools/67.4.0/packages/2c/3a/e0340041dad00234d1fe569ed55b602d24eb11a4939b0a2ec844e970a87d/setuptools-67.4.0-py3-none-any.whl/setuptools-67.4.0.dist-info/METADATA).

Since it is impossible to know how future versions of Python will differ from the current ones, it is equally impossible to specify a "maximum Python version supported".

Please also note that the concept of "working correctly" is something that cannot be "defined a priori". If bugs are discovered in a certain version, setuptools will try to publish a new version with the bug correction, but the old version will still exist, unchanged. So "working" is relative.
